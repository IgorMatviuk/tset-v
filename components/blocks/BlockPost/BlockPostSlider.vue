<template>
  <div class="post-slider">
    <BContainer>
      <div class="post-slider__content">
        <div
          v-swiper:postSlider="swiperOptions"
          :cleanup-styles-on-destroy="false"
          class="post-slider__slider"
        >
          <div class="swiper-wrapper post-slider__slider-wrapper">
            <div
              v-for="(item, index) in items"
              :key="index"
              class="swiper-slide post-slider__slide"
            >
              <BlockPostSliderItem :item="item" />
            </div>
          </div>
          <SliderPagination class="post-slider__pagination" />
          <div class="post-slider__navigation">
            <SliderArrow arrow-type="prev" />
            <SliderArrow arrow-type="next" />
          </div>
        </div>
      </div>
    </BContainer>
  </div>
</template>

<script>
import BlockPostSliderItem from './BlockPostSliderItem'
import SliderPagination from '@/components/shared/SliderPagination'
import SliderArrow from '@/components/shared/SliderArrow'

export default {
  components: {
    BlockPostSliderItem,
    SliderPagination,
    SliderArrow
  },
  props: {
    items: {
      type: Array,
      required: true
    }
  },
  data() {
    return {
      swiperOptions: {
        spaceBetween: 15,
        watchOverflow: true,
        pagination: {
          el: '.post-slider__pagination',
          clickable: true
        },
        navigation: {
          prevEl: '.post-slider__navigation .swiper-button-prev',
          nextEl: '.post-slider__navigation .swiper-button-next'
        },
        keyboard: {
          enabled: true,
          onlyInViewport: true
        },
        breakpoints: {
          768: {
            slidesPerView: 2,
            spaceBetween: 30
          }
        }
      }
    }
  }
}
</script>

<style lang="scss">
.post-slider {
  margin-bottom: 25px;

  &__content {
    margin-right: -15px;
    margin-left: -15px;

    @media only screen and (min-width: 540px) {
      margin-right: 0;
      margin-left: 0;
    }
  }

  &__pagination.swiper-pagination {
    position: static;
    margin-top: 8px;
  }

  &__navigation {
    display: none;
  }

  @include media-breakpoint-up(md) {
    margin-bottom: 4px;

    &__content {
      width: 450px;
      margin-right: auto;
      margin-left: auto;
    }

    &__pagination {
      display: none;
    }

    &__navigation {
      display: flex;
      justify-content: space-between;
      height: 40px;
      margin-right: -5px;
      margin-left: -5px;

      .swiper-button-prev,
      .swiper-button-next {
        position: static;
        margin-top: 0;
      }
    }
  }

  @include media-breakpoint-up(lg) {
    &__content {
      width: 610px;
    }
  }

  @include media-breakpoint-up(xl) {
    margin-bottom: 12px;

    &__content {
      width: 770px;
    }
  }

  @include media-breakpoint-up(xxl) {
    &__content {
      width: 930px;
    }
  }
}
</style>
