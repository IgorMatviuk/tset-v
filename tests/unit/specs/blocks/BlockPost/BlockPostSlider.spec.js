import { mount, createLocalVue } from '@vue/test-utils'
import _cloneDeep from 'lodash.clonedeep'

import { BContainer } from 'bootstrap-vue'
import VueAwesomeSwiper from 'vue-awesome-swiper'

import BlockPostSlider from '@/components/blocks/BlockPost/BlockPostSlider'

import createFactory from '@/api/helpers/factory'
import postAttrs from '@/api/factories/post'

const localVue = createLocalVue()
localVue.use(VueAwesomeSwiper)
localVue.component('BContainer', BContainer)

describe('BlockPostSlider', () => {
  let sliderItems

  function createBlockPostSlider() {
    const wrapper = mount(BlockPostSlider, {
      localVue,
      propsData: {
        items: _cloneDeep(sliderItems)
      },
      mocks: {
        $i18n: {
          path: (url) => url
        },
        $t: (path) => path
      },
      stubs: ['NuxtLink', 'svg-icon']
    })

    return wrapper
  }

  beforeEach(() => {
    const post = createFactory(postAttrs, 1)
    const sliderBlock = post[0].blocks.find((item) => item.type === 'slider')

    sliderItems = sliderBlock.content.items
  })

  test('render posts slider', () => {
    const wrapper = createBlockPostSlider()

    expect(wrapper.element).toMatchSnapshot()
  })
})
