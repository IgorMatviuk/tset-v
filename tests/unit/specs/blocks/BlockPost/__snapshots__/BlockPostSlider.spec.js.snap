// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockPostSlider render posts slider 1`] = `
<div
  class="post-slider"
>
  <div
    class="container"
  >
    <div
      class="post-slider__content"
    >
      <div
        class="post-slider__slider swiper-container swiper-container-initialized swiper-container-horizontal"
      >
        <div
          class="swiper-wrapper post-slider__slider-wrapper"
          style="webkit-transition-duration: 0ms; transition-duration: 0ms; webkit-transform: translate3d(NaNpx, 0px, 0px); transform: translate3d(NaNpx, 0px, 0px);"
        >
          <div
            class="swiper-slide post-slider__slide"
          >
            <div
              class="post-slider-item"
            >
              <div
                class="post-image"
              >
                <div
                  class="img-loader"
                >
                  <picture>
                    <source
                      data-srcset="/img/posts/slider_preview_xl.webp"
                      media="(min-width: 1500px)"
                      srcset="/img/stub.png"
                      type="image/webp"
                    />
                     
                    <source
                      data-srcset="/img/posts/slider_preview_xl.jpg"
                      media="(min-width: 1500px)"
                      srcset="/img/stub.png"
                    />
                     
                    <source
                      data-srcset="/img/posts/slider_preview_lg.webp"
                      media="(min-width: 1200px)"
                      srcset="/img/stub.png"
                      type="image/webp"
                    />
                     
                    <source
                      data-srcset="/img/posts/slider_preview_lg.jpg"
                      media="(min-width: 1200px)"
                      srcset="/img/stub.png"
                    />
                     
                    <source
                      data-srcset="/img/posts/slider_preview_sm.webp"
                      media="(min-width: 992px)"
                      srcset="/img/stub.png"
                      type="image/webp"
                    />
                     
                    <source
                      data-srcset="/img/posts/slider_preview_sm.jpg"
                      media="(min-width: 992px)"
                      srcset="/img/stub.png"
                    />
                     
                    <source
                      data-srcset="/img/posts/slider_preview_xs.webp"
                      media="(min-width: 768px)"
                      srcset="/img/stub.png"
                      type="image/webp"
                    />
                     
                    <source
                      data-srcset="/img/posts/slider_preview_xs.jpg"
                      media="(min-width: 768px)"
                      srcset="/img/stub.png"
                    />
                     
                    <source
                      data-srcset="/img/posts/slider_preview_md.webp"
                      srcset="/img/stub.png"
                      type="image/webp"
                    />
                     
                    <img
                      alt="image title"
                      class="post-image__picture lazyload"
                      data-object-fit="cover"
                      data-object-position="center"
                      data-src="/img/posts/slider_preview_md.jpg"
                      loading="lazy"
                      src="/img/stub.png"
                      title="image title"
                    />
                  </picture>
                </div>
              </div>
            </div>
          </div>
          <div
            class="swiper-slide post-slider__slide"
          >
            <div
              class="post-slider-item"
            >
              <div
                class="post-image"
              >
                <div
                  class="img-loader"
                >
                  <picture>
                    <source
                      data-srcset="/img/posts/slider_preview_xl.webp"
                      media="(min-width: 1500px)"
                      srcset="/img/stub.png"
                      type="image/webp"
                    />
                     
                    <source
                      data-srcset="/img/posts/slider_preview_xl.jpg"
                      media="(min-width: 1500px)"
                      srcset="/img/stub.png"
                    />
                     
                    <source
                      data-srcset="/img/posts/slider_preview_lg.webp"
                      media="(min-width: 1200px)"
                      srcset="/img/stub.png"
                      type="image/webp"
                    />
                     
                    <source
                      data-srcset="/img/posts/slider_preview_lg.jpg"
                      media="(min-width: 1200px)"
                      srcset="/img/stub.png"
                    />
                     
                    <source
                      data-srcset="/img/posts/slider_preview_sm.webp"
                      media="(min-width: 992px)"
                      srcset="/img/stub.png"
                      type="image/webp"
                    />
                     
                    <source
                      data-srcset="/img/posts/slider_preview_sm.jpg"
                      media="(min-width: 992px)"
                      srcset="/img/stub.png"
                    />
                     
                    <source
                      data-srcset="/img/posts/slider_preview_xs.webp"
                      media="(min-width: 768px)"
                      srcset="/img/stub.png"
                      type="image/webp"
                    />
                     
                    <source
                      data-srcset="/img/posts/slider_preview_xs.jpg"
                      media="(min-width: 768px)"
                      srcset="/img/stub.png"
                    />
                     
                    <source
                      data-srcset="/img/posts/slider_preview_md.webp"
                      srcset="/img/stub.png"
                      type="image/webp"
                    />
                     
                    <img
                      alt="image title"
                      class="post-image__picture lazyload"
                      data-object-fit="cover"
                      data-object-position="center"
                      data-src="/img/posts/slider_preview_md.jpg"
                      loading="lazy"
                      src="/img/stub.png"
                      title="image title"
                    />
                  </picture>
                </div>
              </div>
            </div>
          </div>
          <div
            class="swiper-slide post-slider__slide"
          >
            <div
              class="post-slider-item"
            >
              <div
                class="post-image"
              >
                <div
                  class="img-loader"
                >
                  <picture>
                    <source
                      data-srcset="/img/posts/slider_preview_xl.webp"
                      media="(min-width: 1500px)"
                      srcset="/img/stub.png"
                      type="image/webp"
                    />
                     
                    <source
                      data-srcset="/img/posts/slider_preview_xl.jpg"
                      media="(min-width: 1500px)"
                      srcset="/img/stub.png"
                    />
                     
                    <source
                      data-srcset="/img/posts/slider_preview_lg.webp"
                      media="(min-width: 1200px)"
                      srcset="/img/stub.png"
                      type="image/webp"
                    />
                     
                    <source
                      data-srcset="/img/posts/slider_preview_lg.jpg"
                      media="(min-width: 1200px)"
                      srcset="/img/stub.png"
                    />
                     
                    <source
                      data-srcset="/img/posts/slider_preview_sm.webp"
                      media="(min-width: 992px)"
                      srcset="/img/stub.png"
                      type="image/webp"
                    />
                     
                    <source
                      data-srcset="/img/posts/slider_preview_sm.jpg"
                      media="(min-width: 992px)"
                      srcset="/img/stub.png"
                    />
                     
                    <source
                      data-srcset="/img/posts/slider_preview_xs.webp"
                      media="(min-width: 768px)"
                      srcset="/img/stub.png"
                      type="image/webp"
                    />
                     
                    <source
                      data-srcset="/img/posts/slider_preview_xs.jpg"
                      media="(min-width: 768px)"
                      srcset="/img/stub.png"
                    />
                     
                    <source
                      data-srcset="/img/posts/slider_preview_md.webp"
                      srcset="/img/stub.png"
                      type="image/webp"
                    />
                     
                    <img
                      alt="image title"
                      class="post-image__picture lazyload"
                      data-object-fit="cover"
                      data-object-position="center"
                      data-src="/img/posts/slider_preview_md.jpg"
                      loading="lazy"
                      src="/img/stub.png"
                      title="image title"
                    />
                  </picture>
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="swiper-pagination post-slider__pagination"
        />
         
        <div
          class="post-slider__navigation"
        >
          <div
            class="swiper-button-prev"
          >
            <svg-icon-stub
              class="swiper-button-prev-icon"
              height="9"
              name="arrow"
              width="30"
            />
          </div>
           
          <div
            class="swiper-button-next"
          >
            <svg-icon-stub
              class="swiper-button-next-icon"
              height="9"
              name="arrow"
              width="30"
            />
          </div>
        </div>
        <span
          aria-atomic="true"
          aria-live="assertive"
          class="swiper-notification"
        />
      </div>
    </div>
  </div>
</div>
`;
